### PR TITLE
Add course edit feature

### PR DESCRIPTION
**Pull request by LukasAbraham**
- Init the Course details screen with button "Edit" to navigates to the Course Edit Screen.
- The course edit form looks pretty much like the add form with only a few modifications.

### CODE REVIEW

File: src/screens/CoursesScreen.tsx [hunk 1]
```diff
@@ -27,15 +27,15 @@ const CoursesScreen: React.FC<CoursesScreenProps> = ({ navigation }) => {
     try {
       const fetchedCourses = await getAllCourses();
       setCourses(fetchedCourses ?? []);
-    } catch (error) {
-      setError((error as Error).message);
+    } catch (err) {
+      setError((err as Error).message);
```
Comment (maqnitude) [inline src/screens/CoursesScreen.tsx:31]: bruh... you really don't like "error", do you?

**Review by maqnitude — APPROVED** [review]
ok

File: src/screens/CoursesScreen.tsx (hunk 1 above)
Comment (LukasAbraham) [reply src/screens/CoursesScreen.tsx:31]: actually error is already declared as a state in the upper scope, and I really hate seeing the warning message displayed on the screen :))

Comment (maqnitude) [reply src/screens/CoursesScreen.tsx:31]: Shit... my bad. Didn't see any warning for that.

Comment (LukasAbraham) [reply src/screens/CoursesScreen.tsx:31]: it's still gonna work right, it's just that the checker is a bit sensitive

**Pull request merged**